src/alfmarks.php

```diff
@@ -95,12 +95,16 @@ public function score($string) {
 	}
 
 	public function regex() {
-		return $this->regex = $this->regex ?: '/' . implode('|', array_map(function($el) {
-			return implode('|', array_map(function($el) {
-				$word = preg_replace('(.)', '.*?(\0)', $el) . '.*?';
-				return preg_replace('/^(.{3})(.*)(.{3})$/', '\1(\2)\3', $word);
-			}, $el));
-		}, $this->grams())) . '/i';
+		if (!$this->regex) {
+			$this->regex = '/.*?((';
+			$this->regex .= implode(')).*?|.*?((', array_map(function($el) {
+				return implode(').*?(', array_map(function($el) {
+					return preg_quote($el);
+				}, str_split($el)));
+			}, $this->grams()));
+			$this->regex .= ')).*?/i';
```

Comment:
This method is rather complicated, and I couldn't find a good way to simplify it. Maybe that's just the tradeoff when dynamically creating regex?